cf-custom-resources/lib/nlb-cert-validator-updater.js

```diff
@@ -451,6 +457,92 @@ async function unusedValidationOptions(aliases, loadBalancerDNS) {
     return optionsPendingDeletion;
 }
 
+/**
+ * Delete the copilot-tagged certificate that has the default canonical domain name and uses the aliases for its subject
+ * alternative names.
+ * @param {Set<String>} aliases
+ * @returns {Promise<void>}
+ */
+async function deleteCertificate(aliases) {
+    const certificates = await serviceCertificates();
+    const { certPendingDeletion } = categorizeCertificates(aliases, certificates);
+    let promises = [];
+    for (const {CertificateArn: arn} of certPendingDeletion) {
+        let promise = clients.acm().deleteCertificate({
+            CertificateArn: arn
+        }).promise();
+        promises.push(promise);
+    }
+    await Promise.all(promises);
+}
+
+async function deactivate(unusedOptions, loadBalancerDNS, loadBalancerHostedZoneID) {
+    let promises = [];
+    for (let option of unusedOptions) {
+        promises.push(deactivateOption(option, loadBalancerDNS, loadBalancerHostedZoneID));
+    }
+    await Promise.all(promises);
+}
+
+async function deactivateOption(option, loadBalancerDNS, loadBalancerHostedZoneID) {
+    let changes = [{
+        Action: "DELETE",
+        ResourceRecordSet: {
+            Name: option.ResourceRecord.Name,
+            Type: option.ResourceRecord.Type,
+            TTL: 60,
+            ResourceRecords: [
+                {
+                    Value: option.ResourceRecord.Value,
+                }
+            ],
+        }
+    }];
+
+    if (option.Domain !== certificateDomain) {
+        changes.push({
+            Action: "DELETE", // It is validated that if the alias is in use, it is in use by the service itself.
+            ResourceRecordSet: {
+                Name: option.DomainName,
+                Type: "A",
+                AliasTarget: {
+                    DNSName: loadBalancerDNS,
+                    EvaluateTargetHealth: true,
+                    HostedZoneId: loadBalancerHostedZoneID,
+                }
+            }
+        });
+    }
+
+    let {hostedZoneID, route53Client} = await domainResources(option.DomainName);
+    let changeResourceRecordSetsInput = {
+        ChangeBatch: {
+            Comment: `Delete the validation record and A record for ${option.DomainName}`,
+            Changes: changes,
```

Comment:
what's the max batch size, do we feel confident that we won't reach an error because we exceeded the max limit?

Reply:
We do one `changeResourceRecordSets` call for each validation option; each call contains at most two changes in the batch: 1. one change is to remove the validation record; 2. one change is to remove the A-record of the alias.

Since there are at most two changes in each change batch, we won't max out! (Fairly sure that 2 can't be the cap)

Reply:
derp 🤦